cmd/ci-operator/main.go

```diff
@@ -1212,3 +1228,34 @@ func eventRecorder(kubeClient *coreclientset.CoreV1Client, namespace string) rec
 	return eventBroadcaster.NewRecorder(
 		templatescheme.Scheme, coreapi.EventSource{Component: namespace})
 }
+
+func getSSHSecretFromPath(sshKeyPath string) (*coreapi.Secret, error) {
+	secret := &coreapi.Secret{Data: make(map[string][]byte)}
+	secret.Type = coreapi.SecretTypeSSHAuth
+
+	b, err := ioutil.ReadFile(sshKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read file %s for secret: %v", sshKeyPath, err)
+	}
+
+	secret.Name = fmt.Sprintf("ssh-%s", getHashFromBytes(b))
+
+	// Secret.Data["ssh-privatekey"] is required on SecretTypeSSHAuth type.
+	// https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5466-L5470
+	secret.Data[coreapi.SSHAuthPrivateKey] = b
+
+	return secret, nil
+}
+
+func getHashFromBytes(b []byte) string {
```

Comment:
In general I do not like these methods. They are used once and only once, they are not broken out for testing and they break up the flow of the original code. 

We should inline them into the original method unless we are very specifically breaking them out to test them. In both cases -- `getHashFromBytes` and `getSecretName` -- I do not think that the logic inside requires tests.

Reply:
Yea, it makes sense. I will address that.